### ISSUE TITLE
xmonad segfaults!

### ISSUE THREAD

**Issue by l29ah** [xmonad/xmonad#328]
### Problem Description

Outrageous!

### Steps to Reproduce

I don't know, todays master segfaults in a few seconds worth of running time, each time. Reverting to eb48bb4aef638ba6958a4489a282c274e9779767 fixes it. The backtrace seems useless.

### Configuration File

Oh god please no. https://github.com/l29ah/xmonad-config

### Checklist

  - [ ] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - I tested my configuration
    - [x] With `xmonad` version 5da25c5
    - [x] With `xmonad-contrib` version 0c6fdf4

**Comment by liskin** [comment]
If that's an actual segfault, that seems quite unlikely to be caused by a change in Haskell code…
Can you perhaps try to bisect this to a single commit that introduces the crash?

> Oh god please no. https://github.com/l29ah/xmonad-config

Why no?

**Comment by geekosaur** [comment]
I'd also ask what version of GHC; there are some crashing bugs in 8.10.5 that are fixed in 8.10.7, for example.

**Comment by l29ah** [comment]
How can i ask xmonad to use the library that i just built instead of the system-wide installed one?

ghc 9.0.1

**Comment by l29ah** [comment]
Okay, in fact it also reproduces in earlier xmonad revisions. gdb says:
```
#0  0x0000000000d99869 in xmonadzm0zi16zi99999zm1NVWfzz55W6QFvQzzDgdWm3D_XMonadziCore_zdfLayoutClassLayoutWord64zuzdchandleMessage_info ()
```
And there's minimal reproducing config:
```
import XMonad
import XMonad.Layout.Fullscreen

main = xmonad $ fullscreenSupport def
```
Also it seems to only trigger when i have tkabber running.

**Comment by liskin** [comment]
Interesting indeed. Before we attempt to reproduce it, though, may I ask a couple other questions:

1. did you just start using tkabber, which started triggering the crashes, or is this a result of a system upgrade or something?
2. if this is a result of system upgrade, what packages did get upgraded and to what versions?
3. even if not, it'd be good to know what versions of dependencies you have (and what distro and architecture and so on), because a SEGV is quite unlikely to be caused by any Haskell code in xmonad, and is more likely to be an issue in the X11 bindings or perhaps a memory corruption in Xlib or somewhere that happens to crash some time later

**Comment by l29ah** [comment]
No, i'm using tkabber for ages, but i relatively recently (50 days ago) started using fullscreenSupport, and some time later switched to xmonad master to accommodate for ghc 9.0.1.
I don't know if it's a result of system upgrade as my system upgrades each night automatically, much more frequently than i recompile&restart xmonad.
Which dependencies? Gentoo GNU/Linux on amd64 here, x11-libs/libX11 1.7.2, dev-haskell/x11 1.10.

**Comment by l29ah** [comment]
Also now that i think about it, tkabber is quite unique as it spawns >50 windows here, so might be just the sheer amount of windows blowing through some buffer; will need to check for that.

**Comment by l29ah** [comment]
Also sometimes the segfault is triggered by itself a couple of seconds after xmonad starts, sometimes it needs some workspace switching.

**Comment by l29ah** [comment]
And when it segfaults right away, seems like it will segfault right away at the next xmonad start.

**Comment by laMudri** [comment]
This sounds a bit like a problem I've been having for a long time (maybe a year or so) with Zoom, which I've tended to think has something to do with fullscreen. Segfaults tend to happen when switching workspaces to a Zoom window.

**Comment by liskin** [comment]
I've never seen my xmonad crash with Zoom, nor with anything else, so we may need a bit more details. As @geekosaur suggested in IRC today, testing with a different GHC version may be worth it, as 9.0.1 has bad reputation.
If this is still reproducible with other GHC versions, we should definitely look into it, but we'll need a reliable reproducer. It will sound a bit crazy but it worked well in the past: if you can reproduce this in a VM somehow and then send that VM to me, I can boot it in QEMU and debug it there.

**Comment by turboMaCk** [comment]
I remember having issues with xmonad crashing (can't recall if it was segfault) several years ago. In my case it turned out to be problem with manage hooks which lead to dbus messages piling up in some buffer Unfortunately I don't remember details. Anyway my recommendation would be to check manage hooks. You can also try if `XMonad.Hooks.EwmhDesktops` doesn't fix the issue.

**Comment by duplode** [comment]
Though this is a completely anecdotal report (I don't have anything like a proper reproduction), I'll post it just in case it turns out to be of any use. After spending most of 2021 with development versions of *xmonad* and *xmonad-contrib* from January 2021  (at [this](https://github.com/xmonad/xmonad/commit/e363c44bb0694c0932aab7091434c9eeb6e19653) and [this](https://github.com/xmonad/xmonad-contrib/commit/b963c3cf9d3be4f91d84e06c349ca856d8b719dc) commit, respectively; [here](https://github.com/duplode/xmonad-duplode/tree/bde1923f06bc1a9efaffe2b3508c2dd1d91e2759) is my configuration at the time), I started getting segfaults shortly after (though, oddly, not immediately after) upgrading both packages to 0.17. After upgrading *xmobar* from 0.37 to 0.41, the problem appears to have gone away (ten days and counting, while before I didn't get even five minutes without a segfault).

**Comment by geekosaur** [comment]
ghc 9.0.1 is covered by the GHC RTS bug bgamari and I are trying to track down which leads to `nextEvent` smashing the heap. See https://github.com/xmonad/xmonad/issues/389.
